app/controllers/admin/conversations_controller.rb

```diff
@@ -14,11 +15,23 @@ def index
   def show
   end
 
+
+  ## REQUIRE dispatch permissions on the ride zone the conversation is attached to before
+  ## permitting use of close and blacklisting
+  
   # POST /admin/conversations/1/close
+  
+  # user_signed_in?
   def close
-    @conversation.close(current_user.name)
-    flash[:notice] = "Conversation closed."
-    redirect_back(fallback_location: root_path) and return
+    @ride_zone = @conversation.ride_zone
+    
+    if user_signed_in? && has_zone_dispatch?
```

Comment:
Generally I avoid exposing knowledge of endpoints, including their existence, if someone tries to access them without permissions. Bit academic since the source is available.

Another problem with these is that 404s and other errors are false--most of them actually return a 200 http code. I'd like to fix that post-election, but don't want to rock the boat that much prior to it.